bazel_tools/scala.bzl

```diff
@@ -141,27 +141,59 @@ def _strip_path_upto(path, upto):
         return []
 
 def _scala_source_jar_impl(ctx):
-    zipper_args = [
-        "%s=%s" % (new_path, src.path)
-        for src in ctx.files.srcs
-        for new_path in _strip_path_upto(src.path, ctx.attr.strip_upto)
-    ]
-    zipper_args_file = ctx.actions.declare_file(
-        ctx.label.name + ".zipper_args",
-    )
-
     manifest_file = ctx.actions.declare_file(
         ctx.label.name + "_MANIFEST.MF",
-        sibling = zipper_args_file,
     )
     ctx.actions.write(manifest_file, "Manifest-Version: 1.0\n")
-    zipper_args += ["META-INF/MANIFEST.MF=" + manifest_file.path + "\n"]
 
-    ctx.actions.write(zipper_args_file, "\n".join(zipper_args))
+    zipper_args_file = ctx.actions.declare_file(
+        ctx.label.name + ".zipper_args",
+        sibling = manifest_file,
+    )
+    tmpsrcdir = None
+
+    zipper_args = ["META-INF/MANIFEST.MF=" + manifest_file.path]
+    for src in ctx.files.srcs:
+        if src.is_source:
+            for new_path in _strip_path_upto(src.path, ctx.attr.strip_upto):
+                zipper_args.append("%s=%s" % (new_path, src.path))
+        else:
+            if not tmpsrcdir:
+                tmpsrcdir = ctx.actions.declare_directory(ctx.label.name + "_tmpdir")
+            ctx.actions.run(
```

Comment:
I'm not sure this does the right thing.
- Why `inputs = ctx.files.srcs`. As I understand it, this is only ever acting on individual sources, so `inputs = [src]` seems like what you'd want.
- `outputs = [tmpsrcdir]`: Have you tested this with multiple `.srcjar` `srcs`? I'd expect it to fail, because you'd end up with multiple actions producing the same output.
- `if src.is_source`: [Returns true if this is a source file, i.e. it is not generated](https://docs.bazel.build/versions/master/skylark/lib/File.html#is_source). It seems like this would fail for generated `.scala` files, or non-generated `.srcjar` files. Maybe it's better to test `if src.extension == "srcjar"`?

Reply:
Thanks @aherrmann-da.
@nicholassmith-da, could you take care of it in another PR please?

Reply:
I'll create a scenario to check with multiple ```.srcjar``` sources.

My hypothesis is that the ```is_source``` returns false for anything that is read from 'outside' of the Bazel platform and false for all files that are targets. Using the extension is of course fine assuming that all rules that package generated sources do use the ```.srcjar``` extension.

WDYT?

Reply:
Yes, what I meant was if you'd have a generated `.scala` file. E.g.
```
genrule(
  name = "example",
  outs = ["Example.scala"],
  cmd = "echo XYZ > $(OUTS)",
)
```
then `is_source` should be `False` for it, as it's generated, but it's not a `.srcjar`. Similarly, if for whatever reason we'd check in a `.srcjar` into version control and refer to it in `srcs` then `is_source` would return `True` on it even though it's a `.srcjar`. Right now this is probably a purely theoretical issue, as no such case seems to exist in the daml repo.

Indeed, relying on the file extension makes the assumption that the file extension is set correctly.